source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -1,28 +1,76 @@
 #include "extensions/filters/common/local_ratelimit/local_ratelimit_impl.h"
 
+#include "common/protobuf/utility.h"
+
 namespace Envoy {
 namespace Extensions {
 namespace Filters {
 namespace Common {
 namespace LocalRateLimit {
 
-LocalRateLimiterImpl::LocalRateLimiterImpl(const std::chrono::milliseconds fill_interval,
-                                           const uint32_t max_tokens,
-                                           const uint32_t tokens_per_fill,
-                                           Event::Dispatcher& dispatcher)
+LocalRateLimiterImpl::LocalRateLimiterImpl(
+    const std::chrono::milliseconds fill_interval, const uint32_t max_tokens,
+    const uint32_t tokens_per_fill, Event::Dispatcher& dispatcher,
+    const Envoy::Protobuf::RepeatedPtrField<
+        envoy::extensions::common::ratelimit::v3::LocalRateLimitDescriptor>& descriptors,
+    bool per_route)
     : fill_interval_(fill_interval), max_tokens_(max_tokens), tokens_per_fill_(tokens_per_fill),
       fill_timer_(fill_interval_ > std::chrono::milliseconds(0)
                       ? dispatcher.createTimer([this] { onFillTimer(); })
-                      : nullptr) {
+                      : nullptr),
+      time_source_(dispatcher.timeSource()) {
   if (fill_timer_ && fill_interval_ < std::chrono::milliseconds(50)) {
     throw EnvoyException("local rate limit token bucket fill timer must be >= 50ms");
   }
 
-  tokens_ = max_tokens;
+  tokens_.tokens = max_tokens;
 
   if (fill_timer_) {
     fill_timer_->enableTimer(fill_interval_);
   }
+
+  if (per_route) {
+    for (const auto& descriptor : descriptors) {
+      // If there is no limit or descriptors configured, then there is no config
+      // to rate limit on.
+      if (descriptor.entries().empty() || !descriptor.has_token_bucket()) {
+        throw EnvoyException(
+            "wrong config supplied for local rate limit. Either descriptor entries "
+            "or token bucket are missing.");
+        continue;
+      }
```

Comment:
Can this happen? Isn't this blocked by validations? Please double check you have coverage for all error handling, and remove error handling that can't happen.

Reply:
This is the validation we added.. it was previously happening in local rate limit filter but we moved parsing here...

Reply:
? You have proto validations that check this, right? How can this logic be hit?

Reply:
sure.. makes sense.. removing these